kernel/kernel-api/src/main/java/io/delta/kernel/internal/commitrange/CommitRangeFactory.java

```diff
@@ -139,7 +147,151 @@ private void logResolvedVersions(long startVersion, Optional<Long> endVersionOpt
         ctx.endBoundaryOpt);
   }
 
-  private long extractLatestVersion(List<FileStatus> deltaFiles) {
-    return FileNames.deltaVersion(ListUtils.getLast(deltaFiles).getPath());
+  private long extractLatestVersion(List<ParsedDeltaData> deltaDatas) {
+    return ListUtils.getLast(deltaDatas).getVersion();
+  }
+
+  private List<ParsedCatalogCommitData> getFileBasedRatifiedCommits() {
+    // Note: currently this is all we allow in CommitRangeBuilder anyway, but in the future that
+    // could change
+    return ctx.logDatas.stream()
+        .filter(logData -> logData instanceof ParsedCatalogCommitData)
+        .map(logData -> (ParsedCatalogCommitData) logData)
+        .filter(deltaData -> deltaData.isFile())
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Lists the _delta_log and combines the found published deltas with the catalog ratified deltas
```

Comment:
IMO we can just say `catalog commits` and not say `ratified commits` everywhere. i.e. we should pick one and stick with it and `catalog commits` seems most readable